src/libraries/System.Reflection.Emit/tests/PersistableAssemblyBuilder/AssemblySaveWithVariousMembersTests.cs

```diff
@@ -151,33 +153,57 @@ public void AddInterfaceImplementationTest()
+            yield return new object[] { new string[] { "TFirst" } };
+        }
+
+        [Theory]
+        [MemberData(nameof(TypeParameters))]
+        public void SaveGenericTypeParametersForAType(string[] typeParamNames)
         {
             using (TempFile file = TempFile.Create())
             {
-                AssemblyBuilder assemblyBuilder = AssemblyTools.PopulateAssemblyBuilderAndSaveMethod(
-                    s_assemblyName, null, typeof(string), out MethodInfo saveMethod);
-                ModuleBuilder mb = assemblyBuilder.DefineDynamicModule("My Module");
-                TypeBuilder tb = mb.DefineType("TestInterface", TypeAttributes.Interface | TypeAttributes.Abstract);
-                string[] typeParamNames = new string[] { "TFirst", "TSecond", "TThird" };
+                TypeBuilder tb = CreateAssemblyAndDefineType(out AssemblyBuilder assemblyBuilder, out MethodInfo saveMethod);
+                MethodBuilder method = tb.DefineMethod("TestMethod", MethodAttributes.Public);
                 GenericTypeParameterBuilder[] typeParams = tb.DefineGenericParameters(typeParamNames);
-                typeParams[0].SetInterfaceConstraints(new Type[] { typeof(IAccess), typeof(INoMethod)});
-                typeParams[1].SetCustomAttribute(new CustomAttributeBuilder(typeof(DynamicallyAccessedMembersAttribute).GetConstructor(
-                    new Type[] { typeof(DynamicallyAccessedMemberTypes) }), new object[] { DynamicallyAccessedMemberTypes.PublicProperties }));
-                typeParams[2].SetBaseTypeConstraint(typeof(EmptyTestClass));
-                typeParams[2].SetGenericParameterAttributes(GenericParameterAttributes.VarianceMask);
+                if (typeParams.Length > 2)
+                {
+                    SetVariousGenericParameterValues(typeParams);
+                }
                 saveMethod.Invoke(assemblyBuilder, new object[] { file.Path });
 
-                Assembly assemblyFromDisk = AssemblyTools.LoadAssemblyFromPath(file.Path);
-                Type testType = assemblyFromDisk.Modules.First().GetTypes()[0];
+                Type testType = AssemblyTools.LoadAssemblyFromPath(file.Path).Modules.First().GetTypes()[0];
+                MethodInfo testMethod = testType.GetMethod("TestMethod");
                 Type[] genericTypeParams = testType.GetGenericArguments();
-                
-                Assert.Equal(3, genericTypeParams.Length);
-                Assert.Equal("TFirst", genericTypeParams[0].Name);
+
+                Assert.True(testType.IsGenericType);
+                Assert.True(testType.IsGenericTypeDefinition);
+                Assert.True(testType.ContainsGenericParameters);
+                Assert.False(testMethod.IsGenericMethod);
+                Assert.False(testMethod.IsGenericMethodDefinition);
+                //Assert.False(testMethod.ContainsGenericParameters); // Not sure why it is true by default
```

Comment:
It seems by design, this was false for methods that are do not have generic parameters and not in generic Type, I think it was true here because the containing type has generic parameters that can be used for the method